lib/context_loader.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const assert = require('assert');
+const is = require('is-type-of');
+const Loader = require('./loader');
+const classLoader = Symbol('classLoader');
+const EXPORTS = Loader.EXPORTS;
+
+class ClassLoader {
+
+  constructor(options) {
+    assert(options.ctx, 'options.ctx is required');
+    const properties = options.properties;
+    this._cache = new Map();
+    this._ctx = options.ctx;
+
+    for (const property in properties) {
+      this.defineProperty(property, properties[property]);
+    }
+  }
+
+  defineProperty(property, values) {
+    Object.defineProperty(this, property, {
+      get() {
+        if (!this._cache.has(property)) {
+          this._cache.set(property, getInstance(values, this._ctx));
```

Comment:
classloader 是 ctx 级别的，应该会被释放吧

fengmk2 <notifications@github.com>于2016年8月4日 周四上午8:58写道：

> In lib/context_loader.js
> <https://github.com/eggjs/egg-loader/pull/5#discussion_r73448066>:
>
> > +  constructor(options) {
> > +    assert(options.ctx, 'options.ctx is required');
> > +    const properties = options.properties;
> > +    this._cache = new Map();
> > +    this._ctx = options.ctx;
> > +
> > +    for (const property in properties) {
> > +      this.defineProperty(property, properties[property]);
> > +    }
> > +  }
> > +
> > +  defineProperty(property, values) {
> > +    Object.defineProperty(this, property, {
> > +      get() {
> > +        if (!this._cache.has(property)) {
> > +          this._cache.set(property, getInstance(values, this._ctx));
>
> 呃，_ctx 会一直被缓存着不释放？
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eggjs/egg-loader/pull/5/files/98f7bcd5a5cae8e6cddbbf037796c850ca4db42d#r73448066>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAWA1ZjXqt96z3m6t7ft4eV6TtQSrPC7ks5qcTkogaJpZM4Jabuw>
> .
>